sdk/tables/Azure.Data.Tables/tests/perf/Infrastructure/TablesPerfTest.cs

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Data.Tables.Tests;
+using Azure.Test.Perf;
+
+namespace Azure.Data.Tables.Performance
+{
+    public abstract class TablesPerfTest : PerfTest<TablePerfOptions>
+    {
+        private const string stringValue = "This is a string";
+        private Guid guid = Guid.NewGuid();
+        private DateTime dt = DateTime.UtcNow;
+        private byte[] binary = new byte[] { 0x01, 0x02, 0x03, 0x04, 0x05 };
+        private static TablesTestEnvironment _environment = new TablesTestEnvironment();
+        private const string PartitionKey = "performance";
+        private const string TableName = "perfTestTable";
+
+        protected TablesTestEnvironment TestEnvironment => _environment;
+
+        protected TableTransactionalBatch GetBatch() => Client?.CreateTransactionalBatch(PartitionKey);
+
+        protected TableClient Client;
```

Comment:
nit:  Did you mean for this to be a readonly property for derived types or did you intent it as a field?